workers/unity/Packages/com.improbable.gdk.buildsystem/Util/BuildSupportChecker.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+public static class BuildSupportChecker
+{
+    private static readonly Dictionary<BuildTarget, string> BuildPlatformSupportModuleDirectoryNames;
+
+    static BuildSupportChecker()
+    {
+        BuildPlatformSupportModuleDirectoryNames = new Dictionary<BuildTarget, string>();
+
+        const string linuxStandaloneSupport = "LinuxStandaloneSupport";
+
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinux] = linuxStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinux64] = linuxStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinuxUniversal] = linuxStandaloneSupport;
+
+#if !UNITY_EDITOR_OSX
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneOSX] = "MacStandaloneSupport";
+#endif
+
+#if !UNITY_EDITOR_WIN
+        const string windowsStandaloneSupport = "WindowsStandaloneSupport";
+
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneWindows] = windowsStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneWindows64] = windowsStandaloneSupport;
+#endif
+    }
+
+    public struct BuildSupportCheckResult
+    {
+        public readonly bool CanBuild;
+        public readonly BuildTarget[] TargetsWithoutBuildSupport;
+
+        public BuildSupportCheckResult(bool canBuild, BuildTarget[] targetsWithoutBuildSupport)
+        {
+            CanBuild = canBuild;
+            TargetsWithoutBuildSupport = targetsWithoutBuildSupport;
+        }
+    }
+
+    public static BuildSupportCheckResult CheckBuildSupport(params BuildTarget[] buildTargets)
+    {
+        var editorDirectory = Directory.GetParent(EditorApplication.applicationPath);
+
+#if UNITY_EDITOR_OSX
```

Comment:
Why conditional compilation instead of a check?

Reply:
Habit :)